src/index.ts

```diff
@@ -325,8 +326,9 @@ export async function main(
     UploadLogger.verbose(`${parameter}`)
   }
 
-  if (buildParams.slug !== '' && !buildParams.slug?.match(/\//)) {
-    logError(`Slug must follow the format of "<owner>/<repo>" or be blank. We detected "${buildParams.slug}"`)
+  const validSlug =  checkSlug(buildParams.slug)
+  if (!validSlug) {
+    throw new Error(`Slug must follow the format of "<owner>/<repo>" or be blank. We detected "${buildParams.slug}"`)
```

Comment:
Question on this....

I use token authentication to upload codecov reports (using `--token` flag) and because the slug (using the `-r` flag) is not being passed through due to the token being present ([source](https://github.com/codecov/uploader/blob/0e6c734109c78ca4ed746a05a1ff9be24b33c361/src/helpers/provider.ts#L18)), this line is now throwing, and failing my builds. Is there an

More context: I am using Google Cloud Build, which I noticed is not one of the [providers](https://github.com/codecov/uploader/tree/main/src/ci_providers), not sure if that helps.

Reply:
Thanks for the excellent investigation, @sorliem. You gave very relevant information and it provided me with a huge head-start on fixing this issue.

`slug` should not be being validated if a token is passed in.

I created #859 from your comment to track this issue. It should be fixed once #857 is merged.

With regards to Google Cloud Build, I opened #858 to track the feature request.

Reply:
Thanks @mitchell-codecov !